src/dbos-executor.ts

```diff
@@ -1130,46 +1159,70 @@ export class DBOSExecutor implements DBOSExecutorContext {
 
         // If the UUID is preset, it is possible this execution previously happened. Check, and return its original result if it did.
         // Note: It is possible to retrieve a generated ID from a workflow handle, run a concurrent execution, and cause trouble for yourself. We recommend against this.
+        let prevResult: R | Error | DBOSNull = dbosNull;
+        const queryFunc = <T>(sql: string, args: unknown[]) =>
+          this.userDatabase.queryWithClient<T>(client, sql, ...args);
         if (wfCtx.presetUUID) {
-          const func = <T>(sql: string, args: unknown[]) => this.userDatabase.queryWithClient<T>(client, sql, ...args);
-          const check: BufferedResult = await this.#checkExecution<R>(func, workflowUUID, funcId);
-          txn_snapshot = check.txn_snapshot;
-          if (check.output !== dbosNull) {
+          const executionResult = await this.#checkExecution<R>(queryFunc, workflowUUID, funcId);
+          prevResult = executionResult.result;
+          txn_snapshot = executionResult.txn_snapshot;
+          if (prevResult !== dbosNull) {
             tCtxt.span.setAttribute('cached', true);
-            tCtxt.span.setStatus({ code: SpanStatusCode.OK });
-            this.tracer.endSpan(tCtxt.span);
-            return check.output as R;
+
+            if (this.debugMode === DebugMode.TIME_TRAVEL) {
+              // for time travel debugging, navigate the proxy to the time of this transaction's snapshot
+              await queryFunc(`--proxy:${executionResult.txn_id ?? ''}:${txn_snapshot}`, []);
+            } else {
+              // otherwise, return/throw the previous result
```

Comment:
Also need to check if the previous result is dbosNull here. If so, we should throw an error "failed to find the recorded output"

Reply:
This code is in a `if prevResult !== dbosNull` block and there's a check below that throws `DBOSDebuggerError` if we're debugging and there is no previous result. Is there something else that needs to be done?

Reply:
Got it -- somehow I missed that block through the code review UI. Looks good to me now